Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -70,8 +70,10 @@ class MessageQueueManager {
     }
 
     func fetchUserMessagesFromRemoteQueue() {
-        Logger.instance.debug(message: "Fetching user messages from remote queue")
-        fetchUserMessages()
+        DispatchQueue.main.async {
```

Comment:
can we please add comment on why this is needed?

Reply:
Done